packages/apptive_grid_heinzelmen/lib/src/data_widget/data_widget.dart

```diff
@@ -259,6 +259,55 @@ class DataWidget extends StatelessWidget {
       );
     }
 
+    if (data is PhoneNumberDataEntity) {
+      return InkWell(
+        onTap: () {
+          (linkLauncher ?? const LinkLauncher()).openWebPage(
+            url: Uri.parse('tel:${data.value}'),
+            openExternally: true,
+          );
+        },
+        child: Text(
+          data.value.toString(),
+          style: (textStyle ?? const TextStyle()).copyWith(
+            color: Theme.of(context).primaryColor,
+            decoration: TextDecoration.underline,
+          ),
+        ),
+      );
+    }
+
+    if (data is EmailDataEntity) {
+      return InkWell(
+        onTap: () {
+          (linkLauncher ?? const LinkLauncher()).openWebPage(
+            url: Uri.parse('mailto:${data.value}'),
+            openExternally: true,
+          );
+        },
+        child: Text(
+          data.value.toString(),
+          style: (textStyle ?? const TextStyle()).copyWith(
+            color: Theme.of(context).primaryColor,
+            decoration: TextDecoration.underline,
+          ),
+        ),
+      );
+    }
+
+    if (data is SignatureDataEntity) {
+      return AspectRatio(
+        aspectRatio: 2,
+        child: data.value != null
```

Comment:
This check is unnecessary as Line 39 already checks if `data.value == null` and returns the appropriate empty builder here

Reply:
done